compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1146,6 +1146,11 @@ object Types {
       case _ => this
     }
 
+    /** if this type is a reference to a class case of an enum, replace it by its first parent */
+    final def widenEnumCase(using Context): Type = this match
+      case tp: (TypeRef | AppliedType) if tp.classSymbol.isAllOf(EnumCase) => tp.parents.head
```

Comment:
Why the restriction to TypeRefs and AppliedTypes? What should happen with TypeVars, annotated types, or aliases?
A more pervasive alternative would be:
```scala
if tp.classSymbol.isAllOf(EnumCase) then tp.parents.head else tp
```